### PR TITLE
`Development`: Add postgres docker compose setup

### PR DESCRIPTION
**Pull request by 4ludwig4**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

As we want to move towards Postgres, we already implemented a Docker container. We also add support for Docker Compose Setups to enable developers to try Postgres with Artemis quickly.

### Description

We streamlined the official naming of _PostgreSQL_ to _Postgres_ as this informal naming is used in the rest of the project, apparently.
We streamlined the common cypress configurations into a common .env file.
We added Artemis Docker Compose Setups for Postgres and updated the Docs.
Therefore we introduced a new Docker Compose section in the administrator section of the Docs.

### Steps for Testing

**Artemis Postgres Prod Setup**
1. change the permissions in your repository according to the [administrator documentation](https://artemis-platform--6759.org.readthedocs.build/en/6759/admin/productionSetupTips.html#docker-compose-setup) for the prod setup
2. run the Artemis Postgres Prod Setup: ``docker compose -f docker/artemis-prod-postgres.yml up``
3. check Artemis
4. shutdown Artemis Postgres Prod Setup: ``docker compose -f docker/artemis-prod-postgres.yml down`` and optionally delete the contents of ``./docker/.docker-data/artemis-postgres-data``

**Artemis Postgres Dev Setup**
1. run the Artemis Postgres Dev Setup: ``docker compose -f docker/artemis-dev-postgres.yml up``
2. check Artemis
3. shutdown and delete the volumes of Artemis Postgres Prod Setup: ``docker compose -f docker/artemis-dev-postgres.yml down -v``

Check that the other Docker Compose setups you are using still work.

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by TheZoker — APPROVED** [review]
Changes LGTM 👍

**Review by jakubriegel — DISMISSED** [review]
Code looks good 👍🏻 
Both local and dev set up work on my machine 👍🏻 

I think merge should be synchronised with #6443. Artemis was much slower than on MySQL for me, but that my be connected to arm vs x64 container.

**Review by TheZoker — APPROVED** [review]

**Comment by 4ludwig4** [comment]
> I think merge should be synchronised with #6443.

@jakubriegel What do you mean with "synchronized"? There shouldn't be any conflicts between the two PRs.

**Comment by jakubriegel** [comment]
> #6443

Just to make sure that this change also supports arm. It is does, then nothing to be done here

**Comment by jakubriegel** [comment]
> > I think merge should be synchronised with #6443.
> 
> @jakubriegel What do you mean with "synchronized"? There shouldn't be any conflicts between the two PRs.

Just to make sure that this change also supports arm. It is does, then nothing more to be done here

**Comment by 4ludwig4** [comment]
@jakubriegel @TheZoker can you please re-approve if my commit to fix the merge conflicts looks good?

**Review by TheZoker — APPROVED** [review]

**Comment by Mtze** [comment]
Please fix the merge conflicts first ;)

**Review by Hialus — APPROVED** [review]
Changes look good to me.
Though there are some merge conflicts in the .env files.

**Comment by 4ludwig4** [comment]
@Mtze I solved the conflicts and also added common cypress environment settings into one common file.

@TheZoker @Hialus could you please re-review the changes?! Thanks in advance.

**Review by jakubriegel — CHANGES_REQUESTED** [review]
Step 2 (`docker compose -f docker/artemis-prod-postgres.yml up`) starts `artemis-app` without any port exposed, so it is impossible to reach the app from web browser. Most like some config is missing here: https://github.com/ls1intum/Artemis/pull/6759/files#diff-ba5a738c18007d79118321b9c8bf8ac4fe0d466e59605735174862c97d8bf5c5R6

**Comment by 4ludwig4** [comment]
> Step 2 (`docker compose -f docker/artemis-prod-postgres.yml up`) starts `artemis-app` without any port exposed, so it is impossible to reach the app from web browser. Most like some config is missing here: https://github.com/ls1intum/Artemis/pull/6759/files#diff-ba5a738c18007d79118321b9c8bf8ac4fe0d466e59605735174862c97d8bf5c5R6

Mhh maybe I should add in the docs that the ``prod`` setups are served via ``nginx`` on Port 80/443 and the ``dev`` setups are served without a reverse proxy on Port 8080?

Port 8080 is exposed in the inherited ``artemis.yml``.

**Comment by 4ludwig4** [comment]
Fixed the merge conflicts again.

File: .bamboo/E2E-tests/execute.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ if [ "$DB" = "mysql" ]; then
 elif [ "$DB" = "postgresql" ]; then
   COMPOSE_FILE="cypress-E2E-tests-postgres.yml"
 else
-  echo "Invalid database type. Please choose either mysql or postgresql."
+  echo "Invalid database type. Please choose either mysql or postgres."
```
Comment (b-fein) [inline .bamboo/E2E-tests/execute.sh:10]:
```suggestion
  echo "Invalid database type. Please choose either mysql or postgresql."
```
The if-condition above checks for `postgresql`.

**Review by b-fein — APPROVED** [review]

**Review by Hialus — DISMISSED** [review]
Went through the code again and it still looks good to me 🚀

**Review by b-fein — DISMISSED** [review]

**Review by Hialus — DISMISSED** [review]

**Comment by 4ludwig4** [comment]
Fixed the merge conflicts again.

**Review by b-fein — APPROVED** [review]

**Review by Hialus — APPROVED** [review]

**Review by Mtze — APPROVED** [review]

**Pull request merged**